044-Pentagon_numbers.py

```diff
@@ -0,0 +1,34 @@
+# -*- coding: utf-8 -*-
+'''
+Pentagonal numbers are generated by the formula, Pn=n(3n−1)/2. The first ten
+pentagonal numbers are:
+
+1, 5, 12, 22, 35, 51, 70, 92, 117, 145, ...
+
+It can be seen that P4 + P7 = 22 + 70 = 92 = P8. However, their difference,
+70 − 22 = 48, is not pentagonal.
+
+Find the pair of pentagonal numbers, Pj and Pk, for which their sum and
+difference are pentagonal and D = |Pk − Pj| is minimised; what is the value of D?
+'''
+
+
+def pentagonal_numbers():
+    # set can't contain duplicates, is unordered
+    pentagonal_nums = set()
+    n = 1
+
+    while True:
+        n += 1
+        # their sum is pentagonal
+        sum = (3 * n ** 2 - n) / 2
```

Comment:
Avoid using reserved keywords as variable names. The syntax highlight on "sum" is confusing to readers.